controller/payment_handler.go

```diff
@@ -0,0 +1,59 @@
+package controller
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/cyneptic/letsgo/internal/core/service"
+	"github.com/labstack/echo/v4"
+)
+
+type PaymentHandlers struct {
+	srv service.PaymentService
+}
+
+func NewPaymentHandler(srv service.PaymentService) *PaymentHandlers {
+	return &PaymentHandlers{
+		srv: srv,
+	}
+}
+
+func RegisterPaymentRoutes(e *echo.Echo, srv service.PaymentService) {
+	handler := NewPaymentHandler(srv)
+	e.GET("/create-payment", handler.CreatePayment)
+	e.POST("/verify-payment", handler.VerifyPayment)
+}
+
+func (p *PaymentHandlers) CreatePayment(e echo.Context) error {
+	reservationId := e.QueryParam("reservation_id")
+	if reservationId == "" {
+		return echo.ErrBadRequest
+	}
+	// id := e.Get("id").(string)
+	id := "4324902342384"
+	redirectLink, err := p.srv.CreateNewPayment(reservationId, id)
+	if err != nil {
+		return errors.New("error")
+	}
+	return e.HTML(http.StatusTemporaryRedirect, redirectLink)
+
+}
+
+func (p *PaymentHandlers) VerifyPayment(e echo.Context) error {
+	refID := e.FormValue("RefId")
+	resCode := e.FormValue("ResCode")
+	SaleReferenceId := e.FormValue("SaleReferenceId")
+	if resCode != "0" {
+		return http.ErrAbortHandler
+	}
+	reservationId := e.FormValue("SaleOrderId")
```

Comment:
we need validations for all parameters, client might not use what we expect.
validations should be in a controller/validators/ folder in a relative *_validator.go